### PR TITLE
listeners: moving listeners to extension directory

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low (listener code is still hard-coded in due to config validations server)
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
part of https://github.com/envoyproxy/envoy-mobile/issues/2711

### CODE REVIEW

**Comment by alyssawilk** [comment]
(circle ci failure can be ignored)

**Comment by alyssawilk** [comment]
@ggreenway ping?

File: source/server/config_validation/server.cc [hunk 1]
```diff
@@ -102,9 +102,11 @@ void ValidationInstance::initialize(const Options& options,
   Configuration::InitialImpl initial_config(bootstrap_);
   initial_config.initAdminAccessLog(bootstrap_, *this);
   admin_ = std::make_unique<Server::ValidationAdmin>(initial_config.admin().address());
-  listener_manager_ = std::make_unique<ListenerManagerImpl>(
-      *this, std::make_unique<ValidationListenerComponentFactory>(*this), *this, false,
-      quic_stat_names_);
+  listener_manager_ =
+      Config::Utility::getAndCheckFactoryByName<ListenerManagerFactory>(
+          "envoy.listener_manager_impl.default")
```
Comment (ggreenway) [inline source/server/config_validation/server.cc:107]: Make this a constant shared between server/server.cc and here?

File: source/server/config_validation/server.h [hunk 2]
```diff
@@ -24,11 +24,13 @@
 #include "source/common/runtime/runtime_impl.h"
 #include "source/common/secret/secret_manager_impl.h"
 #include "source/common/thread_local/thread_local_impl.h"
+
+// TODO(alyssawilk) address
+#include "source/extensions/listener_managers/listener_manager/listener_manager_impl.h"
```
Comment (ggreenway) [inline source/server/config_validation/server.h:29]: Sill this compile without that extension compiled in? Ideally, config validation would work without the listener manager for all non-listener config.

Comment (alyssawilk) [reply source/server/config_validation/server.h:29]: no this forces the listener extension to be compiled, which is why I have a TODO to fix.

File: docs/root/intro/arch_overview/listeners/listeners.rst [hunk 3]
```diff
@@ -45,3 +45,14 @@ listening on the port. In practice, UDP listeners are configured with the SO_REU
...
+
+.. _extension_envoy.listener_manager_impl.default:
+
+Listener Management
+-------------------
+
+Management of both TCP and UDP listeners is handled by the Listener Manager. For uses of the Envoy
+which do not require any downstream listener, one can build Envoy without the
+``envoy.listener_manager_impl.default`` extension, however this should be done with caution as
+there are other ares of code such as config validation and admin functionality that rely on the listener
```
[the remark this answers is not in the document]
Comment (alyssawilk) [reply docs/root/intro/arch_overview/listeners/listeners.rst:57]: fails on start-up.  I'll rework so it's force built in for envoy server and remove all the docs and comments around it.

**Review by ggreenway — APPROVED** [review]

**Pull request closed without merging**